tonic/models/vic/grid_params.py

```diff
@@ -872,19 +1189,38 @@ def write_netcdf(myfile, target_attrs, target_grid,
             pass
 
         f.createDimension('veg_class', veg_grid['Cv'].shape[0])
-        f.createDimension('root_zone', 3)
+        f.createDimension('root_zone', veg_grid['root_depth'].shape[1])
         f.createDimension('month', 12)
 
+        v = f.createVariable('veg_class', NC_INT, ('veg_class', ))
+        v[:] = np.arange(1, veg_grid['Cv'].shape[0] + 1)
+        v.long_name = 'Vegetation Class'
+
+        v = f.createVariable('veg_descr', np.dtype(str), ('veg_class', ))
+        v[:] = veg_grid['comment']
+        v.long_name = 'Vegetation Class Description'
+
+        v = f.createVariable('root_zone', NC_INT, ('root_zone', ))
+        v[:] = np.arange(1, veg_grid['root_depth'].shape[1] + 1)
+        v.long_name = 'root zone'
+
         v = f.createVariable('month', NC_INT, ('month', ))
         v[:] = np.arange(1, 13)
         v.long_name = 'month of year'
 
         for var, data in veg_grid.items():
+            if var in ['comment']:
+                break
```

Comment:
a couple things here.  

1.  `if var == 'comment'` is the proper comparison.
2.  It may be clearer if you just say `if var != 'comment':`.  This way if `veg_grid` is out of order, you will still write all variables except comment (which I think is what you're going for).